smartmin/users/views.py

```diff
@@ -315,6 +315,7 @@ def form_valid(self, form):
                 FailedLogin.objects.filter(user=user).delete()
                 context['user'] = user
                 context['path'] = "%s" % reverse('users.user_recover', args=[token])
+                context['address'] = self.request.META.get('HTTP_HOST', hostname)
```

Comment:
Why don't we just set this to `hostname` instead of introducing a new subtly different thing. @norkans7 @rowanseymour any downside you see to that?

Reply:
Can see a few different places where we try to get the hostname..

* `settings.HOSTNAME` but default to request.get_host https://github.com/nyaruka/smartmin/blob/6f9636f5c585fb8caa68fd1b12635d6792999b4a/smartmin/users/views.py#L296 and https://github.com/nyaruka/smartmin/blob/2c2a3509a3be90afca354aefe8527ce435d0bce1/smartmin/email.py#L7
* `settings.HOSTNAME` but default to "localhost" 
https://github.com/nyaruka/smartmin/blob/2c2a3509a3be90afca354aefe8527ce435d0bce1/smartmin/templatetags/smartmin.py#L18

Other times `settings.HOSTNAME` is just used as is. I think use that setting, ideally making it required if there's a simple way to do that, or have a consistent backdown.

Reply:
Shmancy linking! I think the problem is that for many sites there is more than one host and that host is important. IE, for U-Report etc, there is no static "hostname" per say because if may be `nigeria.ureport.in` not the global `HOSTNAME` set in settings.py.

Feels pretty safe to me to use the request host since they wouldn't have reached us any other way and if that host wasn't in settings then Django would be blowing up. So just having `hostname` be set to `HTTP_HOST` when doing emails seems "most correct" to me.

Reply:
@nicpottier, in this case, the PR is acceptable with no changes since we're already using `HTTP_POST`.

Reply:
Close, but we should just set `hostname` in the context instead of creating a new variable there `address`.

Reply:
@nicpottier ah ok I thought you were advocating for the setting called `HOSTNAME`. 

Ok that makes sense. Would still be nice to see a consistent way of resolving the host. `request.get_host()` (https://docs.djangoproject.com/en/2.1/ref/request-response/#django.http.HttpRequest.get_host) will use `HTTP_HOST` if it exists so it's probably preferable to looking up the header ourselves.

Reply:
Review again, please.